In [12]:
# Install the plotly library 
pip install plotly

     |████████████████████████████████| 15.6 MB 12.9 MB/s eta 0:00:01
Note: you may need to restart the kernel to use updated packages.


In [17]:
# Import necessary libraries
import pandas as pd
import plotly.express as px
import plotly.graph_objects as go

In [18]:
# Use pandas to read the CSV file and store the data in a DataFrame
df = pd.read_csv('/Users/wanshuo/Desktop/Master/2022:23_I_b/Collecting_Data/week3/Global_Education.csv', encoding='latin-1')
# Display the DataFrame
df

Countries and areas  Latitude    Longitude  OOSR_Pre0Primary_Age_Male  \
0           Afghanistan  33.939110   67.709953                          0   
1               Albania  41.153332   20.168331                          4   
2               Algeria  28.033886    1.659626                          0   
3               Andorra  42.506285    1.521801                          0   
4                Angola  11.202692   17.873887                         31   
..                  ...        ...         ...                        ...   
197           Venezuela   6.423750   66.589730                         14   
198             Vietnam  14.058324  108.277199                          0   
199               Yemen  15.552727   48.516388                         96   
200              Zambia  13.133897   27.849332                          0   
201            Zimbabwe  19.015438   29.154857                         60   

     OOSR_Pre0Primary_Age_Female  OOSR_Primary_Age_Male  \
0                              0                      0   
1                              2                      6   
2                              0                      0   
3                              0                      0   
4                             39                      0   
..                           ...                    ...   
197                           14                     10   
198                            0                      0   
199                           96                     10   
200                            0                     17   
201                           58                      0   

     OOSR_Primary_Age_Female  OOSR_Lower_Secondary_Age_Male  \
0                          0                              0   
1                          3                              6   
2                          0                              0   
3                          0                              0   
4                          0                              0   
..                       ...                            ...   
197                       10                             15   
198                        0                              0   
199                       21                             23   
200                       13                              0   
201                        0                              0   

     OOSR_Lower_Secondary_Age_Female  OOSR_Upper_Secondary_Age_Male  ...  \
0                                  0                             44  ...   
1                                  1                             21  ...   
2                                  0                              0  ...   
3                                  0                              0  ...   
4                                  0                              0  ...   
..                               ...                            ...  ...   
197                               13                             28  ...   
198                                0                              0  ...   
199                               34                             46  ...   
200                                0                              0  ...   
201                                0                             45  ...   

     Primary_End_Proficiency_Reading  Primary_End_Proficiency_Math  \
0                                 13                            11   
1                                  0                             0   
2                                  0                             0   
3                                  0                             0   
4                                  0                             0   
..                               ...                           ...   
197                                0                             0   
198                               55                            51   
199                                0                             0   


In [19]:
# Print basic information about the DataFrame, including column names, non-null counts, etc.
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 202 entries, 0 to 201
Data columns (total 29 columns):
 #   Column                                   Non-Null Count  Dtype  
---  ------                                   --------------  -----  
 0   Countries and areas                      202 non-null    object 
 1   Latitude                                 202 non-null    float64
 2   Longitude                                202 non-null    float64
 3   OOSR_Pre0Primary_Age_Male                202 non-null    int64  
 4   OOSR_Pre0Primary_Age_Female              202 non-null    int64  
 5   OOSR_Primary_Age_Male                    202 non-null    int64  
 6   OOSR_Primary_Age_Female                  202 non-null    int64  
 7   OOSR_Lower_Secondary_Age_Male            202 non-null    int64  
 8   OOSR_Lower_Secondary_Age_Female          202 non-null    int64  
 9   OOSR_Upper_Secondary_Age_Male            202 non-null    int64  
 10  OOSR_Upper_Secondary_Age_Female          202 non-n

# 1. Evaluate the status of tertiary education in different countries and areas

In [20]:
# To calculate the color range
# Obtain the maximum and minimum values of the 'Gross_Tertiary_Education_Enrollment' column
max_value = df['Gross_Tertiary_Education_Enrollment'].max()
min_value = df['Gross_Tertiary_Education_Enrollment'].min()

# Print the maximum and minimum values of the 'Gross_Tertiary_Education_Enrollment' column
print(f"Maximum value: {max_value}")
print(f"Minimum value: {min_value}")

Maximum value: 136.6
Minimum value: 0.0


In [24]:
# Create and display a choropleth map with color mapping
# Use plotly express to create a choropleth map with color mapping, representing the tertiary education enrollment status by countries and areas
fig = px.choropleth(df, 
                    locations='Countries and areas', #Specify the column in the DataFrame that corresponds to the geographical locations (countries and areas)
                    locationmode='country names',#Indicate that the specified locations are represented by country names
                    color='Gross_Tertiary_Education_Enrollment',#The column in the DataFrame used to determine the color of each country on the map 
                    range_color=[0, 140],#Set the color range from 0 to 140, indicating the range of values that will be mapped to colors
                    title='The status of enrollment in tertiary education by countries and areas')#Set the title of the choropleth map

# Center-align the chart title
fig.update_layout(title_text='The status of enrollment in tertiary education by countries and areas', title_x=0.5)
fig.show()# Show the chart

According to the collected data, I chose the column “Gross_Tertiary_Education_Enrollment” as a research object to explore the status of tertiary education in different countries and areas. The color shades present high and low tertiary education enrollment rates, respectively. Darker colors indicate lower enrollment rates, indicating a relatively underdeveloped state of tertiary education in this country or region; Conversely, lighter colors mean higher enrollment rates, reflecting a more developed state of higher education in this country or region.

We can find that the highest tertiary education enrollment is in Australia, at 113.1 through the chart. The dark purple areas mainly focus on developing countries in Africa, such as Congo and Zimbabwe. In contrast, developed countries, for example, the United States and Russia have higher tertiary enrollment rates of 88.2 and 81.9, respectively.

The general trend shows that there is relationship between tertiary education and the level of the economy. The higher the tertiary education enrollment rate is, the more economically developed a country or region is. 

# 2. Analyze the relationship between tertiary education enrollment and the unemployment rate 

In [30]:
# Create a 3D scatter plot using Plotly graph_objects
fig = go.Figure(data=[go.Scatter3d(
    x=df['Gross_Tertiary_Education_Enrollment'],#Extract the 'Gross_Tertiary_Education_Enrollment' column from the DataFrame as x-axis values
    y=df['Unemployment_Rate'],# Extract the 'Unemployment_Rate' column from the DataFrame as y-axis values
    z=df.index, # Use the DataFrame's index as z-axis values, ensuring each point has a unique z-value
    text=df['Countries and areas'],# Use the 'Countries and areas' column for text labels displayed when hovering over points  
    # marker settings:
    mode='markers',
    marker=dict(
        size=8,# Set the size of markers to 8
        color=df['Gross_Tertiary_Education_Enrollment'], # Color the markers based on the 'Gross_Tertiary_Education_Enrollment' values
        colorscale='Viridis', # Set the color scale to Viridis 
        opacity=0.8, # Set the opacity of markers to 0.8
        colorbar=dict(title='Tertiary Education Enrollment') # Add a colorbar with a title for better interpretation of color values
    ),
)])

# scene settings
fig.update_layout(
    scene=dict(
        xaxis_title='Tertiary Education Enrollment',# Set the title for the x-axis
        yaxis_title='Unemployment Rate',# Set the title for the y-axis
        zaxis_title='Index',# Set the title for the z-axis
    ),
    title='3D Relationship between Tertiary Education Enrollment and Unemployment Rate',# Set the overall title of the chart
)

# Display the 3D scatter plot
fig.show()


The X-axis represents tertiary education enrollment, the Y-axis represents the unemployment rate. After excluding some special data (e.g. Greece has a high tertiary education enrollment of 136.6 and a high unemployment rate of 17.24), we can draw a conclusion about the relationship between tertiary education enrollment and unemployment rate from the result of visualization: countries or areas with higher tertiary education enrollment rates generally have lower unemployment rates. Generally speaking, we can clearly observe the trend among the two variables through the three-dimensional visual presentation: higher tertiary education probably is related to lower unemployment rates. 